### DISCUSSION TITLE
Export application details tree as PNG image or any other format

### DISCUSSION THREAD

**Discussion by brunocascio** [argoproj/argo-cd#13892] [Ideas]
We diagram every application by hand (using draw.io or similar tools) with some templates we have, but the information is pretty similar to the ArgoCD details view.
So, would be great if ArgoCD bring us an option of exporting the details view as png or some image format where we can use it afterward in the documentation

**Comment by vbence86** [comment]
I'd like to see this feature too, upvoted 👍

**Comment by Auroch** [comment]
Or exported as open source diagrams code  with K8S nodes
https://diagrams.mingrammer.com/docs/getting-started/examples

**Comment by crenshaw-dev** [comment]
I don't really see this as a first-class feature. For a simple screenshot, one could just use a headless browser. To export to some other tree representation, you could just pull the resource tree from the API and convert it to the desired format.

**Comment by danielpsf-grover** [comment]
Maybe a better ask would be to provide the icon set, similar to https://github.com/kubernetes/community/tree/master/icons or https://aws.amazon.com/architecture/icons/.

What do you think? That would already make a big difference in our lives when drawing diagrams. Then the open-source community could take care of the rest if they want to turn it into a diagram tool.
